app/services/db-export/export-data-files.service.js

```diff
@@ -7,42 +7,44 @@
 
 const path = require('path')
 
-const fs = require('fs').promises
-const os = require('os')
+const fs = require('fs')
+const fsp = fs.promises
 
 /**
  * Writes the converted data to a csv file
  *
  * @param {String} tableConvertedToCsv The converted data to be written to the csv file
  * @param {String} tableName The name of the table
+ * @param {String} schemaFolderPath The path for the schema folder
  *
- * @returns {Boolean} True if the file is written successfully and false if not
+ * @returns {String} Returns the file path of the newly written file
  */
-async function go (tableConvertedToCsv, tableName) {
-  const filePath = _filenameWithPath(tableName)
+async function go (tableConvertedToCsv, tableName, schemaFolderPath) {
+  const filePath = _filenameWithPath(tableName, schemaFolderPath)
 
-  try {
-    await fs.writeFile(filePath, tableConvertedToCsv)
+  await fsp.writeFile(filePath, tableConvertedToCsv)
 
-    return filePath
-  } catch (error) {
-    return false
-  }
+  return filePath
 }
 
 /**
- * Returns a file path by joining the temp directory path with the given file name
+ * Returns a file path by joining the schema folder path with the file name.
+ * The schema path has already been created with the temporary directory
  *
- * @param {String} tableName The name the of the table
+ * @param {String} tableName The name of the table
  *
  * @returns {String} The full file path
  */
-function _filenameWithPath (tableName) {
-  const temporaryFilePath = os.tmpdir()
+function _filenameWithPath (tableName, schemaFolderPath) {
+  const schemaFolderExists = fs.existsSync(schemaFolderPath)
+
+  if (!schemaFolderExists) {
+    fs.mkdirSync(schemaFolderPath)
```

Comment:
I just spotted the docs do seem to mention [`fsPromises.mkdir`](https://nodejs.org/docs/latest-v16.x/api/fs.html#fspromisesmkdirpath-options), if that would work here?